src/canopy/chat_engine/query_generator/condensed.py

```diff
@@ -0,0 +1,139 @@
+import logging
+import re
+from typing import List, Optional, cast
+
+from tenacity import retry, stop_after_attempt, retry_if_exception_type
+
+from canopy.chat_engine.models import HistoryPruningMethod
+from canopy.chat_engine.prompt_builder import PromptBuilder
+from canopy.chat_engine.query_generator import QueryGenerator, LastMessageQueryGenerator
+from canopy.llm import BaseLLM, AnyscaleLLM
+from canopy.models.data_models import Messages, Query, UserMessage
+from canopy.models.api_models import ChatResponse
+
+logger = logging.getLogger(__name__)
+
+SYSTEM_PROMPT = """
+You are an expert on formulating a search query for a search engine, to assist
+ in responding to the user's question.
+
+Given the following conversation, create a standalone question summarizing
+ the user's last question, in its original language.
+
+Reply to me in JSON in this format:
+
+ {"question": {The question you generated here}}.
+
+   Example:
+
+        User: What is the weather today?
+
+        Expected Response:
+            ```json
+            {"question": "What is the weather today?"}
+            ```
+
+    Example 2:
+
+        User: How should I wash my white clothes in the laundry?
+        Assistant: Separate from the colorful ones, and use a bleach.
+        User: Which temperature?
+
+        Expected Response:
+         ```json
+         {"question": "What is the right temperature for washing white clothes?"}
+         ```
+
+
+   Do not try to answer the question; just try to formulate a question representing the user's question.
+   Do not return any other format other than the specified JSON format and keep it really short.
+
+"""  # noqa: E501
+
+USER_PROMPT = "Return only a JSON that has 'question' as a key and the value."
+
+
+class ExtractionException(ValueError):
+    pass
+
+
+class CondensedQueryGenerator(QueryGenerator):
```

Comment:
@igiloh-pinecone what do you think about the name?

Reply:
I tend to agree with @acatav , the name `Instruction` or even `Prompt` is more common.  
The "condensed" naming is a term that LLamaIndex invented, and I haven't seen it used otherwise.